backend/onedrive/onedrive.go

```diff
@@ -78,8 +106,32 @@ func init() {
 		Description: "Microsoft OneDrive",
 		NewFs:       NewFs,
 		Config: func(name string, m configmap.Mapper) {
+			opt := new(Options)
+			err := configstruct.Set(m, opt)
+			if err != nil {
+				fs.Errorf(nil, "Couldn't parse config into struct: %v", err)
+				return
+			}
+
+			graphURL := graphAPIEndpoint + "/v1.0"
+			oauthConfig.Endpoint = *oauthEndpoint
+			switch opt.Region {
+			case "cn":
```

Comment:
Perhaps we can store the graph and auth endpoints in a map (or two maps)? That way we don't need so many switch cases or predefined `oauth2.Endpoint` instances.